### PR TITLE
Fixed feature dev code generated code parsing to handle references

### PR DESCRIPTION
**Pull request by kumsmrit**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description

- Fixed feature dev generated code parsing to handle references to be rendered correctly.
- Missing this fix caused error `Downloaded code results could not be parsed` when generated code results contained references.

## Checklist
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by sarayev — APPROVED** [review]

File: plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/messages/FeatureDevMessagePublisherExtensions.kt [hunk 1]
```diff
@@ -191,11 +192,22 @@ suspend fun MessagePublisher.sendCodeResult(
...
+                repository = ref.repository,
+                url = ref.url,
+                recommendationContentSpan = RecommendationContentSpan(
+                    ref.recommendationContentSpan?.start ?: 0,
+                    ref.recommendationContentSpan?.end ?: 0,
+                ),
+                information = "Reference code under **${ref.licenseName}** license from repository [${ref.repository}](${ref.url})"
+            )
+        }
     }
```
Comment (rli) [inline plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/messages/FeatureDevMessagePublisherExtensions.kt:211]:
```suggestion
    val refs = references.map { ref ->
        CodeReference(
            licenseName = ref.licenseName,
            repository = ref.repository,
            url = ref.url,
            recommendationContentSpan = RecommendationContentSpan(
                ref.recommendationContentSpan?.start ?: 0,
                ref.recommendationContentSpan?.end ?: 0,
            ),
            information = "Reference code under **${ref.licenseName}** license from repository [${ref.repository}](${ref.url})"
        )
    }
```

**Comment by rli** [comment]
need changelog

File: plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/messages/FeatureDevMessagePublisherExtensions.kt (hunk 1 above)
Comment (kumsmrit) [reply plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/messages/FeatureDevMessagePublisherExtensions.kt:211]: We can have `references` returned as emptyList as well from our backend, so we have added the check for if it not empty.

Comment (rli) [reply plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/messages/FeatureDevMessagePublisherExtensions.kt:211]: in another thread, discussed how `map{}` automatically deals with the empty case

**Review by rli — APPROVED** [review]

**Review by dogusata — APPROVED** [review]

**Pull request merged**